api/src/api/userRouter.test.ts

```diff
@@ -391,5 +396,92 @@ describe("userRouter", () => {
         });
       });
     });
+
+    describe("Tax ID changes", () => {
+      it("encrypts the tax id before gets posted", async () => {
+        mockJwt.decode.mockReturnValue(cognitoPayload({ id: "123" }));
+        stubTaxEncryptionDecryptionClient.encryptValue.mockResolvedValue("my cool encrypted value");
+
+        const oldUserData = generateUserData({
+          user: generateUser({ id: "123" }),
+        });
+
+        const updatedUserData = generateUserData({
+          ...oldUserData,
+          profileData: generateProfileData({
+            ...oldUserData.profileData,
+            taxId: "123456789123",
+            encryptedTaxId: undefined,
+          }),
+        });
+
+        stubUserDataClient.get.mockResolvedValue(oldUserData);
+        stubUserDataClient.put.mockResolvedValue(updatedUserData);
+
+        await request(app).post(`/users`).send(updatedUserData).set("Authorization", "Bearer user-123-token");
+
+        expect(stubUserDataClient.put).toHaveBeenCalledWith({
+          ...updatedUserData,
+          profileData: {
+            ...updatedUserData.profileData,
+            taxId: "*******89123",
+            encryptedTaxId: "my cool encrypted value",
+          },
+        });
+      });
+
+      it("masks tax id after encryption", async () => {
+        mockJwt.decode.mockReturnValue(cognitoPayload({ id: "123" }));
+
+        const oldUserData = generateUserData({
+          user: generateUser({ id: "123" }),
+        });
+
+        const updatedUserData = generateUserData({
+          ...oldUserData,
+          profileData: generateProfileData({
+            ...oldUserData.profileData,
+            taxId: "123456789123",
+          }),
+        });
+
+        stubUserDataClient.get.mockResolvedValue(oldUserData);
+        stubUserDataClient.put.mockResolvedValue(updatedUserData);
+        stubTaxEncryptionDecryptionClient.encryptValue.mockResolvedValue("some-encrypted-value");
+
+        await request(app).post(`/users`).send(updatedUserData).set("Authorization", "Bearer user-123-token");
+
+        expect(stubUserDataClient.put).toHaveBeenCalledWith({
+          ...updatedUserData,
+          profileData: {
+            ...updatedUserData.profileData,
+            taxId: "*******89123",
+          },
```

Comment:
what is the point of this test? isn't it testing the same thing as the one above it?